src/main/java/Deadline.java

```diff
@@ -0,0 +1,18 @@
+public class Deadline extends Task{
+
+    protected String by;
+
+    public Deadline(boolean isDone, String taskName, String by) {
+        super(isDone, taskName);
+        this.by = by;
+    }
+
+    public Deadline updateIsDone(boolean newIsDone) {
+        return new Deadline(newIsDone, this.taskName, this.by);
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by + ")";
+    }
+}
```

Comment:
I like the immutability of the updateIsDone method and the clear toString representation. A small suggestion would be to validate the by field if it's supposed to follow a specific format (like a date or time). You might also want to add a method to update the deadline (by) in the future, just like you did for isDone. Otherwise, everything looks really well-structured!